src/app/files/file-explorer.js

```diff
@@ -441,35 +441,42 @@ fileExplorer.prototype.toGist = function (id) {
     }
   }
 
+  async function getOriginalFiles (id) {
+    if (!id) {
+      return []
+    }
+    const url = `https://api.github.com/gists/${id}`
+    const res = await fetch(url)
+    const data = await res.json()
+    return data.files || []
+  }
+
   this.packageFiles(this.files, 'browser/gists/' + id, (error, packaged) => {
     if (error) {
       console.log(error)
       modalDialogCustom.alert('Failed to create gist: ' + error)
     } else {
+      // check for token
       var tokenAccess = this._deps.config.get('settings/gist-access-token')
       if (!tokenAccess) {
         modalDialogCustom.alert(
           'Remix requires an access token (which includes gists creation permission). Please go to the settings tab to create one.'
         )
       } else {
-        var description = 'Created using remix-ide: Realtime Ethereum Contract Compiler and Runtime. \n Load this file by pasting this gists URL or ID at https://remix.ethereum.org/#version=' +
-          queryParams.get().version +
-          '&optimize=' +
-          queryParams.get().optimize +
-          '&gist='
-        var gists = new Gists({
-          token: tokenAccess
-        })
+        const description = 'Created using remix-ide: Realtime Ethereum Contract Compiler and Runtime. \n Load this file by pasting this gists URL or ID at https://remix.ethereum.org/#version=' +
+          queryParams.get().version + '&optimize=' + queryParams.get().optimize + '&gist='
+        const gists = new Gists({ token: tokenAccess })
+
         if (id) {
-          const fileList = Object.keys(this.files.origGistFiles)
-          const updatedFileList = Object.keys(packaged)
+          let originalFileList = getOriginalFiles(id)
           // Telling the GIST API to remove files
-          const allItems = fileList
+          const updatedFileList = Object.keys(packaged)
+          const allItems = Object.keys(originalFileList)
             .filter(fileName => updatedFileList.indexOf(fileName) === -1)
             .reduce((acc, deleteFileName) => ({
               ...acc,
               [deleteFileName]: null
-            }), this.files.origGistFiles)
+            }), originalFileList)
```

Comment:
I've read an article that says that it's actually not a good idea to use spead operator (...) inside a reduce as it's increase the complexity of the function from On to On² (spread operator iterate on each element). Here, the amount of elements is small so it's ok.
You could do that to improve it : 
```typescript
reduce((acc, deleteFileName) => {
  delete acc[deleteFileName]
  return acc
}, originalFileList)
```